go/chat/supersedes.go

```diff
@@ -131,57 +121,59 @@ func (t *basicSupersedesTransform) Run(ctx context.Context,
 			}
 		}
 	}
-	if len(superMsgIDs) == 0 {
-		return originalMsgs, nil
-	}
-
-	var deleteHistoryUpto chat1.MessageID
 
 	// Get superseding messages
-	msgs, err := t.messagesFunc(ctx, conv, uid, superMsgIDs)
-	if err != nil {
-		return nil, err
-	}
-	for _, m := range msgs {
-		if m.IsValid() {
-			supers, err := utils.GetSupersedes(m)
-			if err != nil {
-				continue
-			}
-			for _, super := range supers {
-				smap[super] = m
-			}
+	var deleteHistoryUpto chat1.MessageID
```

Comment:
Add comment about this `if len(superMsgIDs)` case. That we should keep going to clear out retention-deleted messages.